elastic/beats
# OOTB support AWS Eventbridge

## Proposed commit message

Adding OOTB support AWS Eventbridge generated events for S3 changes, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html


## Checklist


## Disruptive User Impact

The logic is triggered as a fall-through and is best effort. 

## Author's Checklist

- [x] Check for idiomatic golang constructs not being used, with golang not being my primary lang there's a good chance for non-idiomatic constructs to sneak in.

## How to test this PR locally

~Since it relies on an AWS setup, testing locally beyond units tests is not possible.~
See /beats/x-pack/filebeat/input/awss3/_meta/terraform/README.md Don't have localstack so not truly 'local' but can run against aws.


## Related issues

As it was discovered as part of support ticket there should be an ER linked.

## Use cases

AWS has added a new way to notify on S3 changes, next to the existing plain SQS and SNS->SQS, AWS now also support using AWS [EventBridge](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html) -> SQS 

This PR achieves two goals:
- no longer need a custom script (which is the current workaround
- Improve user experience by allowing it to 'work out of the box'
